### PR TITLE
[tf.data] graduate experimental `scan` API to tf.data.Dataset

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR graduates the `tf.data.experimental.scan` API into `tf.data.Dataset.scan` by making the following changes:

- [x] Adds the deprecation decorator for the experimental API.
- [x] Add the `scan()` method to `DatasetV2` class.
- [x] Updates example in documentation with new API.
- [x] Regenerate golden API's.
- [x] Moved and updated the `scan_test` target from experimental/kernel_tests to kernel_tests
- [x] Updated the RELEASE.md file

TEST LOG
```
INFO: Build completed successfully, 5 total actions
//tensorflow/python/data/kernel_tests:scan_test                          PASSED in 5.0s
```

cc: @jsimsa @aaudiber

### CODE REVIEW

File: tensorflow/python/data/ops/dataset_ops.py [hunk 1]
```diff
@@ -2675,6 +2675,35 @@ def random(seed=None):
...
+    Args:
+      initial_state: A nested structure of tensors, representing the initial
+        state of the accumulator.
+      scan_func: A function that maps `(old_state, input_element)` to
+        `(new_state, output_element)`. It must take two arguments and return a
+        pair of nested structures of tensors. The `new_state` must match the
+        structure of `initial_state`.
+
+    Returns:
+      A `Dataset` transformation function, which can be passed to
```
Comment (aaudiber) [inline tensorflow/python/data/ops/dataset_ops.py:2702]: Now it just returns "A `Dataset`". A couple of the other recently-migrated transformations in this file have the same issue.

**Review by aaudiber — COMMENTED** [review]
Thanks @kvignesh1420

File: tensorflow/python/data/ops/dataset_ops.py (hunk 1 above)
Comment (kvignesh1420) [reply tensorflow/python/data/ops/dataset_ops.py]: Addresses in the commit. Fixed the `Returns` value for `group_by_window` and `bucket_by_sequence_length` as well.

**Review by aaudiber — APPROVED** [review]

**Comment by kvignesh1420** [comment]
@aaudiber  any update on this? Let me know if additional fixes are required.

**Pull request merged**